pallets/liquidity-mining/src/tests/invariants.rs

```diff
@@ -886,3 +900,404 @@ fn invariant_4() {
             .unwrap();
     });
 }
+
+fn planned_yielding_periods() -> impl Strategy<Value = PeriodOf<Test>> {
+	1_000u64..5_000_000u64
+}
+
+fn percentage() -> impl Strategy<Value = u64> {
+	1..40u64
+}
+
+fn min_deposit() -> impl Strategy<Value = Balance> {
+	1000..10000000u128
+}
+
+#[test]
+//Update global farm for invariant one
+//https://www.notion.so/Liquidity-mining-spec-b30ccfe470a74173b82c3702b1e8fca1#87868f45e4d04ecb92374c5f795a493d
+
+fn update_global_farm_invariant_1() {
+	//Number of sucessfull test cases that must execute for the test as a whole to pass.
+	let successfull_cases = 1_000;
+	//Number of blocks added to current block number in each test case run. This number should be
+	//reasonable smaller than total of runned test to make sure lot of claims is executed and
+	//multiple claims for same deposit to happen.
+	let blocks_offset_range = 1..10_u64;
+	//Index of deposit in `deposit` vec. This idx is used in each test case run and execute claim
+	//if deposit exits.
+
+	let deposit_idx_range = 0..500_usize;
+
+	invariants_externalities().execute_with(|| {
+		let mut runner = TestRunner::new(Config {
+			cases: successfull_cases,
+			source_file: Some("liquidity-mining/src/tests/invariants.rs"),
+			test_name: Some("update_global_farm_invariant_first"),
+			..Config::default()
+		});
+		let deposits: RefCell<Vec<Deposit>> = RefCell::new(Vec::new());
+		//NOTE: farm ids start with 1 so skip 0, farm's id is used as index into array
+		let distributed_before_update: RefCell<[u128; 4]> = RefCell::new([0, 0, 0, 0]);
+
+		runner
+			.run(
+				&(
+					arb_deposit2(),
+					blocks_offset_range,
+					deposit_idx_range,
+					planned_yielding_periods(),
+					percentage(),
+				),
+				|(d, blocks_offset, deposit_idx, planned_yielding_period, percent)| {
+					deposits.borrow_mut().push(d.clone());
+
+					let yield_per_period = Perquintill::from_percent(percent);
+
+					//Act
+					let _ = with_transaction(|| {
+						LiquidityMining::update_global_farm(
+							d.global_farm_id,
+							planned_yielding_period,
+							yield_per_period,
+							1_000,
+						)
+						.unwrap();
+						TransactionOutcome::Commit(DispatchResult::Ok(()))
+					});
+					let g_farm = LiquidityMining::global_farm(d.global_farm_id).unwrap();
+					distributed_before_update.borrow_mut()[d.global_farm_id as usize] =
```

Comment:
```suggestion
//NOTE: `update_global_farm()` distributes rewards before update so this must be done after function execution but before any interaction with farm.
					distributed_before_update.borrow_mut()[d.global_farm_id as usize] =
```
or something similar so it's clear why it's named  "before update" but value is taken after farm's update